Userland/Games/Sudoku/Board.h

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright (c) 2022, the SerenityOS developers.
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include "Square.h"
+#include <AK/Vector.h>
+
+class Board final {
+
+public:
+    Board();
+    Square* get_square(int x, int y) { return &m_squares[x][y]; };
```

Comment:
Sorry I'm not sure why you're asking? If a board exists it shouldn't be null as you can't create a board without m_squares being populated. If you're asking because I'm returning a pointer rather than an object, it's so we can set_value in SudokuWidget without having to replace the object stored in m_board->m_squares

Reply:
I think they're asking why you're not returning a reference then?

Reply:
I return a pointer rather than a reference as some functions like `SudokuWidget::mouse_to_square` return either nullptr or `Board::get_square`. Is there a benefit to using a reference over a pointer?

Reply:
Returning a reference enforces in the type system that the function won't return nullptr (otherwise you can't tell without looking at the implementation)

Also, you could return Optional<T&> in `mouse_to_square` etc instead of raw pointers now (see #13470)